tests/expectations/metrics/test_metric_providers.py

```diff
@@ -29,17 +30,22 @@
 from great_expectations.expectations.metrics.table_metric_provider import (
     TableMetricProvider,
 )
-from great_expectations.expectations.registry import (
-    _registered_metrics,
-)
+from great_expectations.expectations import registry
+
+
+_REGISTERED_METRICS_ORIGINAL: dict = copy.deepcopy(registry._registered_metrics)
 
 
 def test__base_metric_provider__registration():
     """This tests whether the MetricProvider class registers the correct metrics."""
+    registry._registered_metrics = copy.deepcopy(
```

Comment:
Could you use [monkeypatch setattr](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.MonkeyPatch.setattr)?

My concern here is this doesn't reset this attribute after each test. That means, tests that don't test the registry but depend on it indirectly can fail if one of these tests fails part way through. So, I'd like to see monkeypatch or a fixture that unrolls the change.

Reply:
You can factor this into a fixture with a setup and teardown step. You can save whatever state you need to at the beginning of the fixture and reset it during the teardown.

Reply:
@billdirks Done -- please review.  Thanks.